lib/Parse/ParseStmt.cpp

```diff
@@ -2400,3 +2405,54 @@ ParserResult<CaseStmt> Parser::parseStmtCase(bool IsActive) {
                                !BoundDecls.empty(), UnknownAttrLoc, ColonLoc,
                                Body));
 }
+
+/// stmt-pound-assert:
+///   '#assert' '(' expr (',' string_literal)? ')'
+ParserResult<Stmt> Parser::parseStmtPoundAssert() {
+  SyntaxParsingContext AssertContext(SyntaxContext,
+      SyntaxKind::PoundAssertStmt);
```

Comment:
Statement libSyntax parsing is a bit different from expression parsing. The caller (`parseStmt()`) have already [set up the context for the statement](https://github.com/apple/swift/blob/b01776164569cece2706d048707be37e970ce585/lib/Parse/ParseStmt.cpp#L547). So you should just set the syntax kind of it:
```cpp
SyntaxContext->setCreateSyntax(SyntaxKind::PoundAssertStmt);
```

Reply:
thanks, done!